packages/core-users/db/helpers.js

```diff
@@ -106,38 +109,40 @@ Users.helpers({
     );
     return Users.findOne({ _id: this._id });
   },
-  setUsername(username) {
-    Accounts.setUsername(this._id, username);
+  async setUsername(username) {
+    await dbManager.setUsername(this._id, username);
     return Users.findOne({ _id: this._id });
   },
-  addEmail(email, { verified = false, skipEmailVerification = false } = {}) {
-    Accounts.addEmail(this._id, email, verified);
+  async addEmail(
+    email,
+    { verified = false, skipEmailVerification = false } = {}
+  ) {
+    accountsPassword.addEmail(this._id, email, verified);
     if (!skipEmailVerification) {
-      const { sendVerificationEmail = true } = Accounts._options; // eslint-disable-line
-      if (sendVerificationEmail) {
-        Accounts.sendVerificationEmail(this._id, email);
-      }
+      await accountsPassword.sendVerificationEmail(email);
     }
     return Users.findOne({ _id: this._id });
   },
-  removeEmail(email) {
-    Accounts.removeEmail(this._id, email);
+  async removeEmail(email) {
+    await accountsPassword.removeEmail(this._id, email);
     return Users.findOne({ _id: this._id });
   },
-  updateEmail(email, { verified = false, skipEmailVerification = false } = {}) {
+  async updateEmail(
+    email,
+    { verified = false, skipEmailVerification = false } = {}
+  ) {
     log(
       'user.updateEmail is deprecated, please use user.addEmail and user.removeEmail',
       { level: 'warn' }
     );
-    Accounts.addEmail(this._id, email, verified);
+    accountsPassword.addEmail(this._id, email, verified);
     (this.emails || [])
       .filter(({ address }) => address.toLowerCase() !== email.toLowerCase())
-      .forEach(({ address }) => Accounts.removeEmail(this._id, address));
+      .forEach(({ address }) =>
+        accountsPassword.removeEmail(this._id, address)
```

Comment:
Returns promise, shouldn't we have an await Promise.all here to wait until all emails are removed?